### PR TITLE
infra: move wercker no-exception testing to diff.groovy

### PR DESCRIPTION
**Pull request by nmancus1**
infra: move wercker no-exception testing to diff.groovy

Right now, this PR is only a test to make sure that combined diff/launch .groovy will work as intended for no-exceptions testing.  Once PR for https://github.com/checkstyle/contribution/issues/529 is merged, we can change the repo in `checkout-from` back to checkstyle's contribution repo and review this PR.

### CODE REVIEW

**Comment by romani** [comment]
Failed at
```
Command: ./.ci/wercker.sh no-exception-spotbugs
CS_version: 8.29
SEVNTU_version: 1.38.0
```

**Comment by nmancus1** [comment]
> CS_version: 8.29

This is why it failed, I forgot to remove a testing repo of mine from 'projects` file that contained a text block with `\s`, and 8.29 didn't have support for this escape yet.

File: sevntu-checks/.ci/wercker.sh [hunk 1]
```diff
@@ -28,13 +28,13 @@ no-exception-struts)
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo CS_version: ${CS_POM_VERSION}
   echo SEVNTU_version: ${SEVNTU_POM_VERSION}
-  checkout_from https://github.com/checkstyle/contribution.git
+  checkout_from "-b issue-529 https://github.com/nmancus1/contribution.git"
```
Comment (romani) [inline sevntu-checks/.ci/wercker.sh:31]: nice hack

**Comment by nmancus1** [comment]
Leaving a note here: we should replace all usages of `launch.groovy` with `diff.groovy` in single mode.

**Comment by nmancus1** [comment]
Command: `./.ci/wercker.sh no-exception-struts`
maven command from master: `Running command: mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes=**/apache-struts/**/resources/**/* -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR: ``

Command: `./.ci/wercker.sh no-exception-guava`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR:``

Command: `./.ci/wercker.sh no-exception-hibernate-orm`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR:``

Command: ./.ci/wercker.sh no-exception-spotbugs
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes=**/spotbugs-with-excldues/eclipsePlugin-test/**/*,**/spotbugs-with-excldues/spotbugsTestCases/**/*,**/spotbugs-with-excldues/JSR305-testCases/**/* -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`

maven command from this PR:``
maven command from this PR:``

Command: `./.ci/wercker.sh no-exception-spring-framework`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR:``

Command: `./.ci/wercker.sh no-exception-hbase`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR:``

Command: `./.ci/wercker.sh no-exception-Pmd-elasticsearch-lombok-ast`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes=**/pmd/pmd-java/src/test/**/*,**/pmd/cpd/files/* -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes=**/lombok-ast/test/**/* -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from this PR:``
maven command from this PR:``
maven command from this PR:``

Command: `./.ci/wercker.sh no-exception-alot-of-projects`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes=**/apache-ant/src/tests/**/*,**/apache-ant/src/etc/testcases/ -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`
maven command from master: `mvn -e --batch-mode site -Dcheckstyle.config.location=checks-sevntu-error.xml -Dcheckstyle.excludes= -Dcheckstyle.version=8.29 -Dsevntu-checkstyle.version=1.38.0
`

maven command from this PR:``
maven command from this PR:``
maven command from this PR:``
maven command from this PR:``
maven command from this PR:``
maven command from this PR:``

File: sevntu-checks/.ci/wercker.sh [hunk 2]
```diff
@@ -21,149 +21,164 @@ function checkout_from {
...
   cd .ci-temp/contribution/checkstyle-tester
   sed -i.'' 's/^guava/#guava/' projects-to-test-on.properties
   sed -i.'' 's/#hibernate-orm/hibernate-orm/' projects-to-test-on.properties
-  groovy ./launch.groovy --listOfProjects projects-for-wercker.properties \
-      --config checks-sevntu-error.xml --checkstyleVersion ${CS_POM_VERSION} \
-      --sevntuVersion ${SEVNTU_POM_VERSION}
+  groovy ./diff.groovy --listOfProjects projects-for-wercker.properties \
+      --patchConfig checks-sevntu-error.xml --checkstyleVersion ${CS_POM_VERSION} \
+      --sevntuVersion ${SEVNTU_POM_VERSION}   --allowExcludes \
```
Comment (nmancus1) [inline sevntu-checks/.ci/wercker.sh:80]: Here is usage of `--checkstyleVersion` and `--sevntuVersion`.

Comment (romani) [reply sevntu-checks/.ci/wercker.sh:80]: lets use maven extra options of diff.groovy

**Review by romani — APPROVED** [review]
after merge of contribution PR, this PR need to be updated to use upstream contribution repo

**Review by romani — APPROVED** [review]

**Pull request merged**